src/main/java/io/cdap/plugin/salesforce/plugin/source/batch/util/SalesforceSplitUtil.java

```diff
@@ -49,6 +53,9 @@
 public final class SalesforceSplitUtil {
 
   private static final Logger LOG = LoggerFactory.getLogger(SalesforceSplitUtil.class);
+  
+  public static final Set<String> RETRY_ON_REASON = ImmutableSet.of("Unknown", "InternalServerError",
+                                                                    "ClientInputError", "Timeout");
```

Comment:
Why is this getting duplicated in `SalesforceBulkRecordReader` and here?

Reply:
Do we have a documentation how we considered these as retryable?

Also please use `AsyncExceptionCode.<Code>.name()` instead of hardcoded strings.

Reply:
Done